### PR TITLE
Add file share soft delete rule

### PR DESCRIPTION
**Pull request by jonathanruiz**
## PR Summary

- Created Azure.Storage.FileShareSoftDelete (AZR-000298) in Azure.Storage.Rule.ps1
- Created Azure.Storage.FileShareSoft Delete documentation
- Created rules in Azure.Storage.Tests.ps1. Running into issues with having these tests pass.
- Created storage-H in Resources.Storage.json

## PR Checklist

- [X ] PR has a meaningful title
- [X] Summarized changes
- [X] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [X] Unit tests created/ updated
  - [X] Rule documentation created/ updated
  - [X] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

File: tests/PSRule.Rules.Azure.Tests/Azure.Storage.Tests.ps1 [hunk 1]
```diff
@@ -199,6 +199,34 @@ Describe 'Azure.Storage' -Tag Storage {
...
+            $ruleResult[0].Reason | Should -Not -BeNullOrEmpty;
+            $ruleResult[0].Reason | Should -BeExactly 'A sub-resource of type ''Microsoft.Storage/storageAccounts/fileServices'' has not been specified.';
+
+            # Pass
+            $ruleResult = @($filteredResult | Where-Object { $_.Outcome -eq 'Pass' });
+            $ruleResult | Should -Not -BeNullOrEmpty;
+            $ruleResult.Length | Should -Be 1;
+            $ruleResult.TargetName | Should -BeIn 'storage-G';
+            $ruleResult[0].Reason | Should -Not -BeNullOrEmpty;
+            $ruleResult[0].Reason | Should -BeExactly 'Path properties.shareDeleteRetentionPolicy.enabled: Is set to ''True''.';
```
Comment (BernieWhite) [inline tests/PSRule.Rules.Azure.Tests/Azure.Storage.Tests.ps1:221]: @jonathanruiz You need to add a passing test case into tests/PSRule.Rules.Azure.Tests/Resources.Storage.json. Maybe edit storage-G to include configure a `shareDeleteRetentionPolicy` with `enabled` and `days` set.

**Review by BernieWhite — COMMENTED** [review]
@jonathanruiz Nice progress. You're mostly there. Just need to fix the test and add a change log entry.

**Comment by jonathanruiz** [comment]
@BernieWhite Passed all the tests and ready to merge.

**Review by BernieWhite — APPROVED** [review]
@jonathanruiz Nice work! All good to merge.

**Pull request merged**